pkg/monitoring/rules/rules.go

```diff
@@ -0,0 +1,208 @@
+package rules
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/konveyor/forklift-controller/pkg/monitoring/rules/recordingrules"
+)
+
+const (
+	forkliftPrometheusRuleName = "prometheus-forklift-rules"
+
+	prometheusLabelKey   = "prometheus.forklift.konveyor.io"
+	prometheusLabelValue = "true"
+
+	k8sAppLabelKey     = "app"
+	forkliftLabelValue = "forklift"
+)
+
+func SetupRules(namespace string) error {
+	err := recordingrules.Register(namespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func BuildPrometheusRule(namespace string, ownerRef *metav1.OwnerReference) (*promv1.PrometheusRule, error) {
+	rules, err := operatorrules.BuildPrometheusRule(
+		forkliftPrometheusRuleName,
+		namespace,
+		map[string]string{
+			prometheusLabelKey: prometheusLabelValue,
+			k8sAppLabelKey:     forkliftLabelValue,
+		},
+	)
+	rules.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+	if err != nil {
+		return nil, err
+	}
+
+	return rules, nil
+}
+
+func PatchMonitorinLable(namespace string, clientset *kubernetes.Clientset) (err error) {
+	labelKey := "openshift.io/cluster-monitoring"
+	labelValue := "true"
+
+	ns, err := clientset.CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	if val, exists := ns.Labels[labelKey]; exists && val == labelValue {
+		return
+	}
+
+	payload := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": map[string]string{
+				labelKey: labelValue,
+			},
+		},
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return
+	}
+
+	_, err = clientset.CoreV1().Namespaces().Patch(context.TODO(), namespace, types.MergePatchType, payloadBytes, metav1.PatchOptions{})
```

Comment:
why do we change the namespace?

Reply:
we don't, only the labels on the controller to have the updated one.